src/java.desktop/share/classes/com/sun/media/sound/SF2SoundbankReader.java

```diff
@@ -47,9 +47,7 @@ public Soundbank getSoundbank(URL url)
             throws InvalidMidiDataException, IOException {
         try {
             return new SF2Soundbank(url);
-        } catch (RIFFInvalidFormatException e) {
-            return null;
-        } catch(IOException ioe) {
+        } catch (IOException e) {
```

Comment:
This getSoundbank method is called when we iterate over providers and check which can be used, unfortunately, that code is not ready for the IOException and we return "null" -> the next provider will be checked. This is a kind of compatibility quirk.

Reply:
Doesn't IOException mean the next provider is to be checked?

That is the IOException could be handled in the iterator, its code may need to be updated for that. Then the implementations of `getSoundbank(URL)` could propagate the IOException too and eliminate duplicate code.

`InvalidMidiDataException` isn't caught there, so the iterator code should handle it.

Reply:
It is implemented in a way that if the bundled provider will fail we will try some next provider, but if the user add custom provider then getSoundbank(URL) will throw IOException signalling that something went wrong.

I think InvalidMidiDataException should not be caught since that exception means the sound bank provider started to read the file, recognized that the file is supported, but then found the broken midi data.